### DISCUSSION TITLE
Trying to compile a project - errors

### DISCUSSION THREAD

**Discussion by musicstashall** [ALTaleX531/TranslucentFlyouts#135] [Q&A] [answered]
I'm sorry, I'm a beginner. I added the "wil" library to the project, which eliminated some of the errors. Now the undeclared identifiers remain in the _Effect Helper.hpp_ file: **DWM_SYSTEMBACKDROP_TYPE**, **DWMSBT_NONE**, **DWMSBT_TRANSIENT WINDOW**, **DWMSBT_MAINWINDOW**, **DWMSBT_TABBEDWINDOW**, **DWMWA_SYSTEMBACKDROP_TYPE**; in the file _MenuRendering.cpp_ : **MENU_POPUPITEMKBFOCUS**, **MENU_POPUPITEM_FOCUSABLE**, **MENU_POPUPITEMKBFOCUS**. What to do about it, please tell me. I really want to have a working project and implement my ideas on it. Apparently, there are no other necessary libraries here.

**Comment by ALTaleX531** [comment]
Just use Visual Studio 2022 and install the latest SDK and you'll be fine. Or, define these things yourself according to MSDN.

**Reply by musicstashall** [reply]
Thanks, it helped

**Comment by musicstashall** [comment]
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/55966984/9daaa3d5-037b-4287-b266-8cb78b09e434)
There are still a lot of errors. What should I do about it?

**Reply by ALTaleX531** [reply]
Actually, you can still compile it, can't you? It's a display bug in IntelliSense, you can give feedback to the Visual Studio developers. BTW: Do you really have any relevant development experience with Win32 desktop applications? If not I'd recommend learning first rather than editing the code right away, otherwise it might do harm to your PC.

**Reply by musicstashall** [reply]
I have experience VB.NET and AutoIt. Are you ready to help me further so that the project can compile without errors?

![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/55966984/bb171fc0-20e9-4d7c-8384-b1fd4b189c41)

**Reply by ALTaleX531** [reply]
Wait, I remember the project was able to be compiled directly after opening `TranslucentFlyouts.sln` without any problems. How did you manage to have so many errors?

**Reply by musicstashall** [reply]
I have only compiled TFModern, and TFMain is not compiled, the specified errors are present.

**Reply by ALTaleX531** [reply]
How did you add the wil library, wasn't it through the nuget package?

**Reply by musicstashall** [reply]
I connected it through the project properties: C++/General/Additional Include Directories
Could you connect to my computer remotely and set up this project? There may be some libraries missing, etc.

**Reply by ALTaleX531** [reply]
[install-use-packages-visual-studio](https://learn.microsoft.com/zh-cn/nuget/consume-packages/install-use-packages-visual-studio)
Please try google how to use nuget first, you have so many things to learn that it's not something I can handle.

**Reply by musicstashall** [reply]
I installed the Windows Implementation Libraries in the nuget package browser, updated two packages of other libraries there, but the compilation errors are the same.

**Reply by ALTaleX531** [reply]
That's not possible, I downloaded the project's source code and then could compile it directly after restoring all the nuget packages, you must have made other changes.

**Reply by musicstashall** [reply]
Please watch the video from the screen. What am I doing wrong?
https://drive.google.com/file/d/1R9_tjQKQO4vjs52G-dCUORIkjxyQ4KDw/view?usp=sharing

**Reply by ALTaleX531** [reply]
Why did you disable the precompiled header? Just compile, don't touch any settings you don't know what they do. Btw, please compile windbg-workaround before compiling TFMain.

**Reply by musicstashall** [reply]
You've seen that it turns out to be an error if you don't disable headers. An error on the missing pch.h. Maybe all the problems are just in it? And the windbg-workaround project produces even more errors
https://drive.google.com/file/d/1kWYR_EK_VGFmVseZiWDAO9F5-WycWH0B/view?usp=sharing

**Reply by ALTaleX531** [reply, marked as answer]
https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/984c05e8-92a2-4b1f-b8d3-5d1a9b2a94d0

As the video I uploaded demonstrates, no changes are required.
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/246b002c-6404-4ae6-a56a-ee2462bac909)
I don't know why your Visual Studio has enabled support for C++20 module, turn them off as the image shown above.

**Comment by musicstashall** [comment]
The header error has already disappeared, but the errors remain. I probably need to configure the studio. Please don't leave me)
Compiling:
https://drive.google.com/file/d/1ORPSu7gu-BHEr-6K3xo3CVJT2EjnzjZK/view?usp=sharing
Settings:
https://drive.google.com/file/d/1gGfm-P9MxUJTY2gpmSM9n3vMz7W_sOE9/view?usp=sharing

They told me: don't mess with C++))

**Reply by ALTaleX531** [reply]
I'm sorry, but I can't access the file you shared. Also, I have never configured Visual Studio since I installed it and I don't know how to help you.

**Reply by musicstashall** [reply]
I just decided to upgrade VS 2022 to version 17.9.6 and everything started compiling. Before that, I had version 17.6. Thank you for your patience and help.
